drivers/media/i2c/tc358743.c

```diff
@@ -219,7 +223,7 @@ static noinline void i2c_wrreg(struct v4l2_subdev *sd, u16 reg, u32 val, u32 n)
 
 static u8 i2c_rd8(struct v4l2_subdev *sd, u16 reg)
 {
-	return i2c_rdreg(sd, reg, 1);
+	return i2c_rdreg(sd, reg, 1, NULL);
```

Comment:
This is a bit ugly - why not make i2c_rdreg a call to a new function - i2c_rdreg_err, say, passing in a NULL error pointer? The callers that care about the error would then call the new function.

Reply:
Will do.